### DISCUSSION TITLE
How to start pocketbase with grpc?

### DISCUSSION THREAD

**Discussion by Oggy4th** [pocketbase/pocketbase#3640] [Q&A]
I'm trying to serve echo with grpc service, I found this solution [REST and GRPC in parallel](https://chrysanthium.com/rest-and-grpc-in-parallel) which I don't have access to net.Listener.

Is there any other way to do it? How to start on same port?

**Comment by ganigeorgiev** [comment]
I don't think it is possible at the moment without starting a separate listener on a new port.

I guess one hacky workaround could be to start the new listener with custom low level socket options like `SO_REUSEPORT` but I haven't tried it and I don't think it is supported everywhere.

**Reply by Oggy4th** [reply]
So how to start in different port in production?

**Reply by ganigeorgiev** [reply]
It is the same as starting a regular Go listener, the only difference is that you'll need to run it with `go` prefix to execute in a new goroutine to avoid blocking the statements that follows it, eg. `go http.ListenAndServe(addr, handler)`.

**Reply by ghost** [reply]
Is it possible to use reverse proxy for this?

**Reply by ganigeorgiev** [reply]
@temzints I guess yes, but the go application still will have to fire the grpc listener on a different port.
